packages/fxa-shared/payments/stripe.ts

```diff
@@ -258,37 +266,39 @@ export abstract class StripeHelper {
   }
 
   /**
-   * Append any matching price ids and names to their corresponding AbbrevPlayPurchase.
+   * Append any matching prices and related info to their corresponding IAP purchases.
    */
-  async addPriceInfoToAbbrevPlayPurchases(
-    purchases: AbbrevPlayPurchase[]
+  async addPriceInfoToIapPurchases(
+    purchases: PlayStoreSubscriptionPurchase[] | AppStoreSubscriptionPurchase[],
+    iapType: Omit<SubscriptionType, typeof MozillaSubscriptionTypes.WEB>
   ): Promise<
-    (AbbrevPlayPurchase & {
-      product_id: string;
-      product_name: string;
-      price_id: string;
-    })[]
+    | (
+        | (PlayStoreSubscriptionPurchase | AppStoreSubscriptionPurchase) &
+            IapExtraStripeInfo
+      )[]
```

Comment:
Nit: This seems a little unwieldy, might be better to just pull out the type above and use it by name.